tests/stdlib/tmath.nim

```diff
@@ -358,6 +358,12 @@ template main() =
     doAssert almostEqual(prod([1.5, 3.4]), 5.1)
     let x: seq[float] = @[]
     doAssert prod(x) == 1.0
+  
+  block: # clamp range
+    doAssert clamp(10, 1..5) == 5
+    doAssert clamp(1, 1..3) == 1
+    doAssert clamp(42.0, 1.0 .. 3.1415926535) == 3.1415926535
+    doAssert clamp(-10, -5 .. -1) == -5
```

Comment:
How about adding some more tests, for example
```nim
doAssert clamp(3, 1..5) == 3
doAssert clamp(5, 1..5) == 5
```

Reply:
I've merged this inside suggestion https://github.com/nim-lang/Nim/pull/17246/files#r586572493